core/storage/ipfs/impl/blockservice_impl.cpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/ipfs/impl/blockservice_impl.hpp"
+
+using fc::storage::ipfs::Block;
+
+namespace fc::storage::ipfs {
+  outcome::result<void> BlockServiceImpl::addBlock(const Block &block) {
+    return local_storage_.set(block.getCID(), block.getContent());
+  }
+
+  outcome::result<bool> BlockServiceImpl::has(const CID &cid) const {
+    return local_storage_.contains(cid);
+  }
+
+  outcome::result<Block::Content> BlockServiceImpl::getBlockContent(
+      const CID &cid) const {
+    outcome::result<Block::Content> data = local_storage_.get(cid);
+    return data ? data : BlockServiceError::NOT_FOUND;
```

Comment:
```
if (!data && data.error() == IpfsDatastoreError::NOT_FOUND)
   return BlockServiceError::NOT_FOUND;
else
   return data;
```
Otherwise low-level errors can be hidden.